omod-common/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/MainResourceController.java

```diff
@@ -155,11 +154,19 @@ public Object delete(@PathVariable("resource") String resource, @PathVariable("u
 	@RequestMapping(value = "/{resource}/{uuid}", method = RequestMethod.DELETE, params = "purge")
```

Comment:
I believe changing the value of this **params** attribute is enough without touching the method code

Reply:
That way you don't have to bake the void/retire logic in the method.

Reply:
I think I'm missing something @wluyima . 

>A resource should only be purged when the purge parameter is set to true

The above being the title of this ticket, how would this be possible without changing the method code accordingly. I think I'm not understanding you ..

Reply:
I think the params attribute can take a spring EL expression, I think in this case you need to set the value for the purge parameter to true, you might want to look at the spring docs for the params attribute on RequestMapping for the exact syntax

Reply:
That way spring will only invoke this controller if purge is set explicitly set to true by the client, you won't have to make any code changes in the controller method.